awx/main/utils/receptor.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import yaml
+import time
+
+from receptorctl.socket_interface import ReceptorControl
+
+
+logger = logging.getLogger('awx.main.utils.receptor')
+
+
+def get_receptor_ctl():
+    return ReceptorControl('/var/run/receptor/receptor.sock')
+
+
+def worker_info(node_name):
+    receptor_ctl = get_receptor_ctl()
+    transmit_start = time.time()
+    error_list = []
+    data = {'Errors': error_list, 'transmit_timing': 0.0}
+
+    result = receptor_ctl.submit_work(worktype='ansible-runner', payload='', params={"params": f"--worker-info"}, ttl='20s', node=node_name)
+
+    unit_id = result['unitid']
+    run_start = time.time()
+    data['transmit_timing'] = run_start - transmit_start
+    data['run_timing'] = 0.0
+
+    try:
+
+        resultfile = receptor_ctl.get_work_results(unit_id)
+
+        stdout = ''
+
+        while data['run_timing'] < 20.0:
```

Comment:
maybe leave a comment that `ttl` should probably be >= this value. I can seem someone coming by to increase this timeout, but forget about the ttl in the work submission command

Reply:
hmm.. I'm actually not totally sure if I understand what you're saying. Is it okay if both timeouts are 20 seconds?

What is most _reasonable_? Would it be better for transmit timeout to be 20 seconds, and run timeout to be 40 seconds?

Reply:
yeah it's fine if both are 20 seconds. By transmit timeout you mean `ttl`?

After the receptor job expires due to ttl (20 seconds), it will have a Fail state in receptor. Then in your code the `if state_name` conditional hits and we break the while loop. Therefore, increasing the `while data['run_timing'] < 20.0:` to say, 500 seconds, would still have no real effect because we'll still break the loop after 20 seconds due to the ttl.

As such, it's probably best if we make sure ttl matches this `while data['run_timing']` timeout